src/Microsoft.OpenApi.OData.Reader/Edm/ODataPathProvider.cs

```diff
@@ -273,15 +273,24 @@ private void RetrieveNavigationPropertyPaths(IEdmNavigationProperty navigationPr
             Debug.Assert(navigationProperty != null);
             Debug.Assert(currentPath != null);
 
+            // Check whether the navigation property has already been navigated in the path
+            foreach (ODataSegment segment in currentPath)
```

Comment:
checking on the name only is not sufficient. We could have something completely different in the path with the same name.